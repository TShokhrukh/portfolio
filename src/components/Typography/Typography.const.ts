import { cn } from '@bem-react/classname'

export const cnTypography = cn('Typography')
