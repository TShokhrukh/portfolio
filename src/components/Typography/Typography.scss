.Typography {
  --font-size: inherit;
  --color: inherit;

  font-size: var(--font-size);
  color: var(--color);
}
