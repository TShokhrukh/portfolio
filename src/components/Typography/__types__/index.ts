export * from './Typography.types'

export * from './Typography_variant_body.types'
export * from './Typography_variant_h1.types'
export * from './Typography_variant_h2.types'
export * from './Typography_variant_h3.types'
export * from './Typography_variant_h4.types'
export * from './Typography_variant_h5.types'
export * from './Typography_variant_h6.types'
export * from './Typography_variant_span.types'

export * from './Typography_color_dark.types'
export * from './Typography_color_purple.types'
export * from './Typography_color_rose.types'
export * from './Typography_color_white.types'
