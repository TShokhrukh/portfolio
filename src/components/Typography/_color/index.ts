export * from './Typography_color_dark'
export * from './Typography_color_purple'
export * from './Typography_color_rose'
export * from './Typography_color_white'
