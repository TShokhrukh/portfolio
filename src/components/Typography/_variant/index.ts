export * from './Typography_variant_body'
export * from './Typography_variant_span'
export * from './Typography_variant_h1'
export * from './Typography_variant_h2'
export * from './Typography_variant_h3'
export * from './Typography_variant_h4'
export * from './Typography_variant_h5'
export * from './Typography_variant_h6'
