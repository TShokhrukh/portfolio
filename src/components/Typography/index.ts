export * from './build'
export * from './__types__'
export * from './_color'
export * from './_variant'
export { Typography as TypographyBase } from './Typography'
