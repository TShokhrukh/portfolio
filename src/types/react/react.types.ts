import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
