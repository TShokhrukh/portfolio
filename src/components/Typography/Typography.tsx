import React from 'react'

import { cnTypography } from './Typography.const'
import type { ITypographyProps } from './__types__'

import './Typography.scss'

export const Typography: React.FC<ITypographyProps> = ({
  component: Component = 'p',
  className,
  children,
  ...props
}) => {
  return (
    <Component {...props} className={cnTypography({}, [className])}
      >{children}</Component>
  )
}
