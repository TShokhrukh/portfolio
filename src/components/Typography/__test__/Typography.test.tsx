/* global test, jest, expect, describe */
import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { Typography } from '../Typography'

describe('Typography', () => {
  test('must return the component template', () => {
    render(<Typography data-testid='typography' className='temp_class'>children</Typography>)

    expect(screen.getByTestId('typography')).toMatchSnapshot()
  })

  test('must change the component by "component" prop', () => {
    render(<Typography data-testid='typography' component='div' />)

    expect(screen.getByTestId('typography').tagName).toBe('DIV')
  })
})
