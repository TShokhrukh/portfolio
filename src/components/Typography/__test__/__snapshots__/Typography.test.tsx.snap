// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography must return the component template 1`] = `
<p
  class="Typography temp_class"
  data-testid="typography"
>
  children
</p>
`;
